OneSignalSDK/onesignal/src/main/java/com/onesignal/OSNotificationGenerationJob.java

```diff
@@ -133,6 +131,21 @@ public OSNotification getNotification() {
     }
 
     public void setNotification(OSNotification notification) {
+        // we determine whether an android notification id has been
+        // generated by us only for the first notification. The assumption
+        // is any subsequent notification updates are derived from the
+        // initial one, therefore the didGenerateId will persist across.
+        if(this.notification == null)
```

Comment:
I see where we have a `notDisplayNotificationLogic` method that I don't think will support this logic:
https://github.com/OneSignal/OneSignal-Android-SDK/blob/4.7.3/OneSignalSDK/onesignal/src/main/java/com/onesignal/OSNotificationController.java#L108-L109

A suggestion is create some kind of didDisplay flag which might make this simpler. Either way we probably need to update all points accessing this class and account for display / not display logic.

Reply:
@jkasten2 I reworked the fix to include a `isNotificationToDisplay` flag, and adjusted everyone looking for a -1 Android ID to now look for that flag.

Reply:
Thanks, looks good to me.